Assets/Mirror/Editor/MirrorAssetProcessor.cs

```diff
@@ -0,0 +1,129 @@
+/*
+MIT License
+
+Pull request submitted by Digital Ruby, LLC (https://www.digitalruby.com)
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using UnityEngine;
+using UnityEditor;
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace DigitalRuby.WeatherMaker
+{
+    /// <summary>
+    /// Pre-process define utility
+    /// </summary>
+    public class MirrorAssetPostprocessor : AssetPostprocessor
+    {
+        // add list of asset files to key off of along with pre-processor definition
+        // asset name / pre processor value key / value pair
+        internal static readonly KeyValuePair<string, string>[] preProcessors = new KeyValuePair<string, string>[]
+        {
+            new KeyValuePair<string, string>("/Mirror/Runtime/NetworkIdentity.cs", "MIRROR_NETWORKING_PRESENT"),
+        };
+
+        /// <summary>
+        /// Update pre-processor
+        /// </summary>
+        /// <param name="hasAsset">Whether the asset exists, if it does pre-processor is added, else it is removed if it exists</param>
+        /// <param name="preProcessor">The pre-processor to add or remove</param>
+        internal static void UpdatePreProcessor(bool hasAsset, string preProcessor)
```

Comment:
Split this into 2 separate methods:
AddPreprocessor  and RemovePreprocessor  instead of passing the boolean.   That would follow the single responsibility principle.

Reply:
Ok